xarray/core/duck_array_ops.py

```diff
@@ -76,6 +76,11 @@ def isnull(data):
         return _isnull(data)
     except TypeError:
         return np.zeros(data.shape, dtype=bool)
+    except Exception:
```

Comment:
I'm a little nervous about relying on an exception for checking for null data, because xarray uses this check all over the place, and I'm pretty sure that `_isnull` (which calls `pandas.isnull`) will first try to convert unknown array types into NumPy arrays.

For now, I think this is OK, but I'm going to try to write another helper function for doing this entirely in xarray.

Reply:
See https://github.com/pydata/xarray/pull/3132 for a cleaner fix here.

Reply:
merged into this PR

Reply:
#3132 isn't quite working yet, as evidenced by the failing tests. It would probably be best to leave that for a follow-up :)